src/extension.ts

```diff
@@ -15,26 +15,42 @@
  */
 import * as path from 'path';
 import { commands, Disposable, ExtensionContext, extensions, Terminal, TextDocument, window, workspace } from 'vscode';
-import { registerVSCodeCommands } from './commands/registerCommands';
-import { QuarkusPropertiesLanguageMismatch, QuarkusConfig } from './QuarkusConfig';
+import { registerVSCodeClientCommands, registerVSCodeCommands } from './commands/registerCommands';
+import { QuarkusConfig, QuarkusPropertiesLanguageMismatch } from './QuarkusConfig';
 import { QuarkusContext } from './QuarkusContext';
 import quarkusProjectListener from './QuarkusProjectListener';
 import { connectToQuteLS } from './qute/languageServer/client';
 import { terminalCommandRunner } from './terminal/terminalCommandRunner';
 import { tryToForceLanguageId } from './utils/languageMismatch';
 import { JAVA_EXTENSION_ID } from './utils/requestStandardMode';
 import { initTelemetryService } from './utils/telemetryUtils';
+import { getFilePathsFromWorkspace } from './utils/workspaceUtils';
 import { WelcomeWebview } from './webviews/WelcomeWebview';
 import { createTerminateDebugListener } from './wizards/debugging/terminateProcess';
 
 // alias for vscode-java's ExtensionAPI
 export type JavaExtensionAPI = any;
 
 export async function activate(context: ExtensionContext) {
-
   await initTelemetryService(context);
 
   QuarkusContext.setContext(context);
+
+  registerVSCodeClientCommands(context);
+  // Check if activation occured due to a 'java' language document
+  const onLanguageJavaWasActivated = workspace.textDocuments.some(doc => doc.languageId === 'java');
+  if (onLanguageJavaWasActivated) {
```

Comment:
One issue might arise if you have a rootless project or a single Java file, then you generate a Quarkus project into the file. When you go to try and use any of the other Quarkus features once the project is generated, like debugging, then it won't work.

However, this is more of an edge case, and I think it's worth it to do this the way you've coded it.